compiler/cpp/src/generate/t_java_generator.cc

```diff
@@ -441,7 +441,7 @@ string t_java_generator::java_type_imports() {
 }
 
 string t_java_generator::java_suppressions() {
-  return "@SuppressWarnings({\"cast\", \"rawtypes\", \"serial\", \"unchecked\"})\n";
+  return "@SuppressWarnings(\"all\")\n";
```

Comment:
We fixed the code generator for cpp to generate code without warnings.
Suppress some specific warnings instead of all is preferred as well.